views/default/output/url.php

```diff
@@ -84,5 +82,29 @@
 unset($vars['is_trusted']);
 unset($vars['confirm']);
 
-$attributes = elgg_format_attributes($vars);
-echo "<a $attributes>$text</a>";
+$vars['class'] = elgg_extract_class($vars, 'elgg-anchor');
+
+$text = elgg_format_element('span', [
+	'class' => 'elgg-anchor-label',
+], $text);
+
+$icon = elgg_extract('icon', $vars, '');
+unset($vars['icon']);
+
+if ($icon && !preg_match('/^</', $icon)) {
+	$icon = elgg_view_icon($icon, [
+		'class' => 'elgg-anchor-icon',
+	]);
+}
+
+$indicator = elgg_extract('indicator', $vars);
+unset($vars['indicator']);
+
+if (!is_null($indicator)) {
```

Comment:
On a second thought this can just be a data attribute on the anchor itself. We don't need to add extra markup for it.